assets/src/hooks/useChannel.ts

```diff
@@ -143,3 +143,82 @@ export const useCheckedChannel = <T, U>({
   ])
   return state
 }
+
+export const useCheckedTwoWayChannel = <T, U>({
+  socket,
+  topic,
+  event,
+  dataStruct,
+  parser,
+  loadingState,
+  initialMessage,
+}: {
+  socket: Socket | undefined
+  topic: string | null
+  event: string
+  initialMessage: any
+  dataStruct: Struct<T, any>
+  parser: (data: T) => U
+  loadingState: U
+}): [U, (event: string, payload: any) => void] => {
+  const [state, setState] = useState<U>(loadingState)
+  const [channel, setChannel] = useState<Channel | undefined>()
+
+  const onOk = ({ data: data }: { data: unknown }) => {
+    try {
+      assert(data, dataStruct)
+      setState(parser(data))
+    } catch (error) {
+      if (error instanceof StructError) {
+        Sentry.captureException(error)
+      }
+    }
+  }
+
+  const pushUpdate = useCallback(
+    (event: string, payload: any): void => {
+      if (channel) {
+        channel
+          .push(event, payload)
+          .receive("ok", (data: { data: unknown }) => {
+            onOk(data)
+          })
+      }
+    },
+    [channel]
+  )
+
+  useEffect(() => {
+    setState(loadingState)
+    let channel: Channel | undefined
+    if (socket !== undefined && topic !== null) {
+      channel = initialMessage
+        ? socket.channel(topic, initialMessage)
+        : socket.channel(topic)
+      channel.on(event, (data: { data: unknown }) => {
+        onOk(data)
+      })
+      channel.onError((error) => console.log(error))
+
+      channel
+        .join()
+        .receive("ok", (data: { data: unknown }) => {
+          onOk(data)
+          setChannel(channel)
+        })
+        .receive("error", ({ reason }) =>
+          // eslint-disable-next-line no-console
+          console.error(`joining topic ${topic} failed`, reason)
+        )
+        .receive("timeout", reload)
+    }
+
+    return () => {
+      if (channel !== undefined) {
+        channel.leave()
+        channel = undefined
+      }
+    }
+  }, [socket, topic, event, loadingState, dataStruct, parser])
```

Comment:
*issue:** PSA that I'm getting a React hooks eslint warning about a missing dependency here (`onOk`). Just thought I'd mention it since for now warnings aren't triggering CI failures so it can be easy to miss unless you run `npm check` locally.